packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -2506,55 +2492,132 @@ class RenderPointerListener extends RenderProxyBoxWithHitTestBehavior {
   /// Called when a pointer that triggered an [onPointerDown] changes position.
   PointerMoveEventListener onPointerMove;
 
+  /// Called when a pointer that triggered an [onPointerDown] is no longer in
+  /// contact with the screen.
+  PointerUpEventListener onPointerUp;
+
+  /// Called when the input from a pointer that triggered an [onPointerDown] is
+  /// no longer directed towards this receiver.
+  PointerCancelEventListener onPointerCancel;
+
+  /// Called when a pointer signal occurs over this object.
+  PointerSignalEventListener onPointerSignal;
+
+  @override
+  void performResize() {
+    size = constraints.biggest;
+  }
+
+  @override
+  void handleEvent(PointerEvent event, HitTestEntry entry) {
+    assert(debugHandleEvent(event, entry));
+    // The onEnter, onHover, and onExit events are triggered from within
+    // MouseTracker, not here.
+    if (onPointerDown != null && event is PointerDownEvent)
+      return onPointerDown(event);
+    if (onPointerMove != null && event is PointerMoveEvent)
+      return onPointerMove(event);
+    if (onPointerUp != null && event is PointerUpEvent)
+      return onPointerUp(event);
+    if (onPointerCancel != null && event is PointerCancelEvent)
+      return onPointerCancel(event);
+    if (onPointerSignal != null && event is PointerSignalEvent)
+      return onPointerSignal(event);
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    final List<String> listeners = <String>[];
+    if (onPointerDown != null)
+      listeners.add('down');
+    if (onPointerMove != null)
+      listeners.add('move');
+    if (onPointerUp != null)
+      listeners.add('up');
+    if (onPointerCancel != null)
+      listeners.add('cancel');
+    if (onPointerSignal != null)
+      listeners.add('signal');
+    if (listeners.isEmpty)
+      listeners.add('<none>');
+    properties.add(IterableProperty<String>('listeners', listeners));
+    // TODO(jacobr): add raw listeners to the diagnostics data.
```

Comment:
@jacob314 Is there anything special we need to do to resolve this TODO?

Reply:
```dart
properties.add(ObjectFlagProperty<PointerDownEventListener>('onPointerDown', onPointerDown, ifPresent: 'down', level: DiagnosticLevel.fine);
```
Would be an idiomatic way to expose each of these.
That would hide the output from the default display but let the inspector provide links to the actual listeners used.

Reply:
@jacob314 
The current way allows a clean format with the help of `IterableProperty`,
```
'listeners: down, move, up, cancel, signal'
```
or 
```
'listeners: <none>'
```

Is there a way to maintain the current format while using `ObjectFlagProperty`?

Reply:
Not currently. Are there any cases with a subclass that wants to add an additional listener? In that case it could make sense to come up with a scheme that enables defining each listener as an ObjectFlagProperty and then aggregate the properties into a terse summary like `listeners: down, move, up, cancel` when generating the string output.